compiler/luci/pass/src/FusePReluPass.cpp

```diff
@@ -15,17 +15,173 @@
  */
 
 #include "luci/Pass/FusePReluPass.h"
+#include "helpers/NodeFiller.h"
 
 #include <luci/IR/CircleNodes.h>
 
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/Service/CircleNodeClone.h>
+
+#include <cassert>
+
+// Helper to fuse PRelu
+namespace
+{
+
+/**
+ * Below diagram shows PRelu pattern to fuse.
+ * - this pattern will be replaced with one PRelu
+ *
+ *           [In]
+ *            |
+ *            V
+ *     +---- ifm ----+
+ *     |      |      |
+ *     |      |      V
+ *     |      |     abs
+ *     |      V      |
+ *     |     sub <---+
+ *     |      |
+ *     |      V
+ *     |   mul_alpha (alpha of PRelu)
+ *     |      |
+ *     V      V
+ *    relu mul_half (0.5)
+ *     |      |
+ *     |      V
+ *     +---> add
+ *            |
+ *            V
+ *          [Out]
+ *
+ */
+class PReluPattern final
+{
+public:
+  PReluPattern(luci::CircleAdd *candidate)
+  {
+    assert(candidate);
+    _add_ofm = candidate;
+  }
+
+private:
+  bool match();
+
+public:
+  bool matched();
+
+public:
+  luci::CircleNode *_ifm = nullptr;
+  luci::CircleRelu *_relu = nullptr;
+  luci::CircleAbs *_abs = nullptr;
+  luci::CircleSub *_sub = nullptr;
+  luci::CircleMul *_mul_alpha = nullptr;
+  luci::CircleMul *_mul_half = nullptr;
+  luci::CircleAdd *_add_ofm = nullptr;
+  luci::CircleConst *_const_alpha = nullptr;
+  luci::CircleConst *_const_half = nullptr;
+
+private:
+  bool _matched = false;
+};
+
+#define CHECK_OR_FALSE(condition) \
+  if (not(condition))             \
+    return false;
+
+bool PReluPattern::match()
```

Comment:
CircleMul, CircleSub, CircleAdd can have fused act function.

They have to be None.